playbooks/roles/caasp4/tasks/cleanup_nodes.yml

```diff
@@ -0,0 +1,55 @@
+---
+#
+# (c) Copyright 2019 SUSE Linux GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+# NOTE(toabctl): We import the resources (which are created by the network heat
+# stack in setup_nodes.yml so we need to remove the resources again from
+# terraform state and let heat handle the reource cleanup
+
+- name: Does the terraform state exist
+  stat:
+    path: "{{ socok8s_workspace }}/skuba-terraform/terraform.tfstate"
+  register: _terraform_state_file
+
+
+- name: Remove existing networking resources from terraform state
+  block:
```

Comment:
nit: You don't need a block if you don't want to handle exceptions. Also this will result in a lot of "skipped" tasks, so you can probably do an include_tasks of the whole series of tasks from L28 to L55 when ``_terraform_state_file.stat.exists``. It would dynamically evaluate the include, and bypass some of the exec loop, so it would run faster, and is generally cleaner.

Reply:
Done. Thanks for the hint!